packages/utils/upgrade/resources/codemods/5.0.0/utils-public-interface.code.ts

```diff
@@ -0,0 +1,315 @@
+It will also warn about removed functions to avoid breaking user code.
+
+ESM
+
+Before:
+
+import * as utils from '@strapi/utils';
+
+utils.nameToSlug();
+
+After:
+
+import { strings } from '@strapi/utils';
+
+strings.nameToSlug();
+
+---
+ESM
+
+Before:
+
+import { nameToSlug } from '@strapi/utils';
+
+nameToSlug();
+
+After:
+
+import { strings } from '@strapi/utils';
+
+strings.nameToSlug();
+
+---
+
+Common JS
+
+Before:
+
+const utils = require('@strapi/utils');
+
+utils.nameToSlug();
+
+After:
+
+const { strings } = require('@strapi/utils');
+
+strings.nameToSlug();
+
+---
+Common JS
+
+Before:
+
+const { nameToSlug } = require('@strapi/utils');
+
+nameToSlug();
+
+After:
+
+const { strings } = require('@strapi/utils');
+
+strings.nameToSlug();
+
+*/
+
+const changes = {
+  strings: {
+    nameToSlug: 'nameToSlug',
+    nameToCollectionName: 'nameToCollectionName',
+    stringEquals: 'isEqual',
+    isCamelCase: 'isCamelCase',
+    isKebabCase: 'isKebabCase',
+    toKebabCase: 'toKebabCase',
+    toRegressedEnumValue: 'toRegressedEnumValue',
+    startsWithANumber: 'startsWithANumber',
+    joinBy: 'joinBy',
+  },
+  arrays: {
+    stringIncludes: 'includesString',
+  },
+  objects: {
+    keysDeep: 'keysDeep',
+  },
+  dates: {
+    generateTimestampCode: 'timestampCode',
+  },
+  async: {
+    pipeAsync: 'pipe',
+    mapAsync: 'map',
+    reduceAsync: 'reduce',
+  },
+};
+
+const removed = [
+  'getCommonBeginning',
+  'templateConfiguration',
+  'removeUndefined',
+  'getConfigUrls',
+  'getAbsoluteAdminUrl',
+  'getAbsoluteServerUrl',
+  'forEachAsync',
+];
+
+const transformImports = (root: Collection, j: JSCodeshift) => {
+  root
+    .find(j.ImportDeclaration, {
+      source: { value: '@strapi/utils' },
+    })
+    .forEach((path) => {
+      path.value.specifiers?.forEach((specifier) => {
+        if (!j.ImportSpecifier.check(specifier)) {
+          return false;
+        }
+
+        if (removed.includes(specifier.imported.name)) {
+          console.warn(
+            `Function "${specifier.imported.name}" as removed. You will have to remove it from your code.`
+          );
+
+          return false;
+        }
+      });
+
+      for (const primitive of Object.keys(changes)) {
+        const functions = Object.keys(changes[primitive]);
+
+        const specifiersToRefactor = path.value.specifiers?.filter((specifier) => {
+          return j.ImportSpecifier.check(specifier) && functions.includes(specifier.imported.name);
+        });
+
+        if (specifiersToRefactor?.length > 0) {
+          path.value.specifiers?.unshift(j.importSpecifier(j.identifier(primitive)));
+
+          specifiersToRefactor.forEach((specifier) => {
+            const index = path.value.specifiers?.indexOf(specifier);
+            path.value.specifiers?.splice(index, 1);
+          });
+        }
+      }
+
+      if (path.value.specifiers?.length === 0) {
```

Comment:
That's odd, it would mean that the import/require is already empty, no?

Reply:
it filters out namespace & default specifiers so you could have just a default or a namespace specifier